### PR TITLE
Automated cherry pick of #45: Don't be too shallow

### PR DESCRIPTION
**Pull request by doublek**
Cherry pick of #45 on release-v1.1.

#45: Don't be too shallow

### CODE REVIEW

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Review by Brian-McM — APPROVED** [review]

**Pull request merged**
